src/Adapter/Discount/Update/DiscountUsabilityUpdater.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Discount\Update;
+
+use CartRule;
+use Doctrine\DBAL\Connection;
+use PrestaShop\PrestaShop\Core\Domain\Discount\Command\AddDiscountCommand;
+use PrestaShop\PrestaShop\Core\Domain\Discount\Command\UpdateDiscountCommand;
+
+class DiscountUsabilityUpdater
+{
+    public function __construct(
+        private readonly Connection $connection,
+        private readonly string $dbPrefix
+    ) {
+    }
+
+    /**
+     * Update customer group restrictions for a discount.
+     *
+     * @param CartRule $discount
+     * @param AddDiscountCommand|UpdateDiscountCommand $command
+     *
+     * @return array List of updated properties
+     */
+    public function updateCustomerGroups(CartRule $discount, $command): array
```

Comment:
```suggestion
     * @return array List of updated properties
     */
    public function updateCustomerGroups(CartRule $discount, WriteDiscountCommandInterface $command): array
```
What do you think of adding an interface to AddDiscountCommand and UpdateDiscountCommand so we can strong type the $command parameter ? ( the name of the interface is tbd ) 
If we don't choose that at least add a mixed type and don't delete the phpdoc on this param.

Reply:
I thinkk we could simplify it with an array of discount IDs directly, no need to couple this service with any CQRS commands

Reply:
Same comment, do not couple the service with a command simply for one field of the command

But anyway, I think this one is not used anymore and can be removed completely